src/sentry/static/sentry/app/views/performance/vitalDetail/table.tsx

```diff
@@ -0,0 +1,421 @@
+
+const getTableColumnTitle = (index: number, vitalName: WebVital) => {
+  const abbrev = vitalAbbreviations[vitalName];
+  const titles = [
+    ...COLUMN_TITLES,
+    `${abbrev}(p50)`,
+    `${abbrev}(p75)`,
+    `${abbrev}(p95)`,
+    `${abbrev}(Status)`,
+  ];
+  return titles[index];
+};
+
+export function getProjectID(
+  eventData: EventData,
+  projects: Project[]
+): string | undefined {
+  const projectSlug = (eventData?.project as string) || undefined;
+
+  if (typeof projectSlug === undefined) {
+    return undefined;
+  }
+
+  const project = projects.find(currentProject => currentProject.slug === projectSlug);
+
+  if (!project) {
+    return undefined;
+  }
+
+  return project.id;
+}
+
+type Props = {
+  eventView: EventView;
+  organization: Organization;
+  location: Location;
+  setError: (msg: string | undefined) => void;
+  summaryConditions: string;
+
+  projects: Project[];
+};
+
+type State = {
+  widths: number[];
+};
+
+class Table extends React.Component<Props, State> {
+  state = {
+    widths: [],
+  };
+
+  handleCellAction = (column: TableColumn<keyof TableDataRow>) => {
+    return (action: Actions, value: React.ReactText) => {
+      const {eventView, location, organization} = this.props;
+
+      trackAnalyticsEvent({
+        eventKey: 'performance_views.overview.cellaction',
+        eventName: 'Performance Views: Cell Action Clicked',
+        organization_id: parseInt(organization.id, 10),
+        action,
+      });
+
+      const searchConditions = tokenizeSearch(eventView.query);
+
+      // remove any event.type queries since it is implied to apply to only transactions
+      searchConditions.removeTag('event.type');
+
+      updateQuery(searchConditions, action, column.name, value);
+
+      ReactRouter.browserHistory.push({
+        pathname: location.pathname,
+        query: {
+          ...location.query,
+          cursor: undefined,
+          query: stringifyQueryObject(searchConditions),
+        },
+      });
+    };
+  };
+
+  renderBodyCell(
+    tableData: TableData | null,
+    column: TableColumn<keyof TableDataRow>,
+    dataRow: TableDataRow,
+    vitalName: WebVital
+  ): React.ReactNode {
+    const {eventView, organization, projects, location, summaryConditions} = this.props;
+
+    if (!tableData || !tableData.meta) {
+      return dataRow[column.key];
+    }
+    const tableMeta = tableData.meta;
+
+    const field = String(column.key);
+    const fieldRenderer = getFieldRenderer(field, tableMeta);
+    const rendered = fieldRenderer(dataRow, {organization, location});
+
+    const allowActions = [
+      Actions.ADD,
+      Actions.EXCLUDE,
+      Actions.SHOW_GREATER_THAN,
+      Actions.SHOW_LESS_THAN,
+    ];
+
+    if (field === 'count_unique(user)') {
+      return (
+        <UniqueUserCell>
+          {rendered}
+          <StyledUserIcon size="20" />
+        </UniqueUserCell>
+      );
+    }
+
+    if (field === getVitalDetailTableStatusFunction(vitalName)) {
```

Comment:
Should these special cased fields happen before 131-132? As that work is not needed and is waste.